dao/src/main/scala/za/co/absa/enceladus/menasplugin/MenasCredentials.scala

```diff
@@ -18,13 +18,36 @@ package za.co.absa.enceladus.menasplugin
 import java.io.File
 
 import com.typesafe.config.ConfigFactory
+import org.apache.hadoop.fs.FileSystem
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.SparkSession
+import org.slf4j.LoggerFactory
+import za.co.absa.enceladus.utils.fs.FileSystemVersionUtils
 
 object MenasCredentials {
 
-  def fromFile(path: String): MenasCredentials = {
-    val conf = ConfigFactory.parseFile(new File(replaceHome(path)))
+  private val logger = LoggerFactory.getLogger(this.getClass)
+  
+  /**
+   * Retrieves Menas Credentials from file either on local file system or on HDFS
+   */
+  def fromFile(path: String)(implicit spark: SparkSession): MenasCredentials = {
+    val conf = if(path.startsWith("hdfs://")) {
+     val file = new FileSystemVersionUtils(spark.sparkContext.hadoopConfiguration).hdfsRead(path)
+     ConfigFactory.parseString(file)
+    } else {
+     ConfigFactory.parseFile(new File(replaceHome(path)))
```

Comment:
I like what you did, but I think it could have been taken a step further.

Reply:
Feel free to make sugesstions

Reply:
So I looked it over yesterday, and it would be too big of a change yielding too little benefit if any.